src/Compilers/CSharp/Test/Symbol/Symbols/StaticAbstractMembersInInterfacesTests.cs

```diff
@@ -10946,5 +10966,1580 @@ unsafe class Test
+
+            Assert.Null(c1.FindImplementationForInterfaceMember(m01));
+            Assert.Null(i1.FindImplementationForInterfaceMember(m01));
+
+            var source2 =
+@"
+public class C1 : I1
+{
+   static void I1.M01() {}
+}
+";
+
+            var compilation2 = CreateCompilationWithIL(source2, ilSource, options: TestOptions.DebugDll,
+                                                 parseOptions: TestOptions.RegularPreview,
+                                                 targetFramework: TargetFramework.NetCoreApp);
+
+            compilation2.VerifyEmitDiagnostics(
+                // (4,19): error CS0539: 'C1.M01()' in explicit interface declaration is not found among members of the interface that can be implemented
+                //    static void I1.M01() {}
+                Diagnostic(ErrorCode.ERR_InterfaceMemberNotFound, "M01").WithArguments("C1.M01()").WithLocation(4, 19)
+                );
+
+            c1 = compilation2.GlobalNamespace.GetTypeMember("C1");
+            m01 = c1.Interfaces().Single().GetMembers().OfType<MethodSymbol>().Single();
+
+            Assert.Null(c1.FindImplementationForInterfaceMember(m01));
+        }
+
+        [Fact]
+        public void ImplementAbstractStaticMethod_12()
+        {
+            // Ignore invalid metadata (default interface implementation for a static method)
+
+            var ilSource = @"
+.class interface public auto ansi abstract I1
+{
+    .method public hidebysig abstract virtual 
+        static void M01 () cil managed 
+    {
+    } // end of method I1::M01
+} // end of class I1
+.class interface public auto ansi abstract I2
+    implements I1
+{
+    // Methods
+    .method private hidebysig 
+        static void I1.M01 () cil managed 
+    {
+        .override method void I1::M01()
+        IL_0000: ret
+    } // end of method I2::I1.M01
+} // end of class I2
+";
+
+            var source1 =
+@"
+public class C1 : I2
+{
+}
+";
+
+            var compilation1 = CreateCompilationWithIL(source1, ilSource, options: TestOptions.DebugDll,
+                                                 parseOptions: TestOptions.RegularPreview,
+                                                 targetFramework: TargetFramework.NetCoreApp);
+
+            compilation1.VerifyEmitDiagnostics(
+                // (2,19): error CS0535: 'C1' does not implement interface member 'I1.M01()'
+                // public class C1 : I2
+                Diagnostic(ErrorCode.ERR_UnimplementedInterfaceMember, "I2").WithArguments("C1", "I1.M01()").WithLocation(2, 19)
+                );
+
+            var c1 = compilation1.GlobalNamespace.GetTypeMember("C1");
+            var i2 = c1.Interfaces().Single();
+            var i1 = i2.Interfaces().Single();
+            var m01 = i1.GetMembers().OfType<MethodSymbol>().Single();
+
+            Assert.Null(c1.FindImplementationForInterfaceMember(m01));
+            Assert.Null(i2.FindImplementationForInterfaceMember(m01));
```

Comment:
Should we test a scenario where we attempt to implement an `abstract static` method from metadata that has a default implementation?

Reply:
> Should we test a scenario where we attempt to implement an abstract static method from metadata that has a default implementation?

Perhaps I am missing something, but I do not find this interesting. A presence of default implementation (BTW, it is not allowed for static members) has no impact on an attempt to implement a member.